src/Service/CognitoIdentityProvider/src/CognitoIdentityProviderClient.php

```diff
@@ -220,6 +220,61 @@ public function verifySoftwareToken($input): VerifySoftwareTokenResponse
         return new VerifySoftwareTokenResponse($response);
     }
 
+    protected function getEndpointMetadata(string $region): array
+    {
+        switch ($region) {
+            case 'ap-northeast-1':
+            case 'ap-northeast-2':
+            case 'ap-south-1':
+            case 'ap-southeast-1':
+            case 'ap-southeast-2':
+            case 'ca-central-1':
+            case 'eu-central-1':
+            case 'eu-west-1':
+            case 'eu-west-2':
+            case 'us-east-1':
+            case 'us-east-2':
+            case 'us-west-2':
+                return [
+                    'endpoint' => 'https://cognito-idp.%region%.amazonaws.com',
+                    'signRegion' => $region,
+                    'signService' => 'cognito-idp',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'fips-us-east-1':
+                return [
+                    'endpoint' => 'https://cognito-idp-fips.us-east-1.amazonaws.com',
+                    'signRegion' => 'us-east-1',
+                    'signService' => 'cognito-idp',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'fips-us-east-2':
+                return [
+                    'endpoint' => 'https://cognito-idp-fips.us-east-2.amazonaws.com',
+                    'signRegion' => 'us-east-2',
+                    'signService' => 'cognito-idp',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'fips-us-west-2':
+                return [
+                    'endpoint' => 'https://cognito-idp-fips.us-west-2.amazonaws.com',
+                    'signRegion' => 'us-west-2',
+                    'signService' => 'cognito-idp',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+        }
+
+        throw new \InvalidArgumentException(sprintf('The region "%s" is not supported by "CognitoIdentityProvider".', $region));
```

Comment:
should it be a dedicated exception rather than a generic InvalidArgumentException ?

Reply:
I suggest adding an `UnsupportedRegionException`.

Reply:
thanks for the suggestion